src/sinks/datadog/traces/stats.rs

```diff
@@ -398,36 +406,107 @@ impl Aggregator {
         });
         m
     }
+
+    /// Flushes the bucket cache of stale entries.
+    /// Based on https://github.com/DataDog/datadog-agent/blob/cfa750c7412faa98e87a015f8ee670e5828bbe7f/pkg/trace/stats/concentrator.go#L38-L41
+    /// It means that we cache and can compute stats only for the last `BUCKET_WINDOW_LEN * BUCKET_DURATION_NANOSECONDS` and after such time,
+    /// buckets are then flushed. This only applies to past buckets. Stats buckets in the future are cached with no restriction.
+    fn flush(&mut self) {
+        let now = Utc::now().timestamp_nanos() as u64;
+        let flush_cutoff_time = now - (BUCKET_DURATION_NANOSECONDS * BUCKET_WINDOW_LEN);
+
+        // remove entries from the map if the start time is outside the retaining window
+        self.buckets.retain(|&bucket_start, _bucket| {
+            let flush = bucket_start >= flush_cutoff_time;
+
+            if flush {
+                debug!("Flushing {} start_time bucket.", bucket_start);
+            }
+            flush
```

Comment:
This is technically backwards, because it will only drop the entry if `flush` is false.

Reply:
Good catch. I changed the bool var to represent the behavior of the `retain()` function as well.

Reply:
resolved in https://github.com/vectordotdev/vector/pull/14694/commits/82d0e97b9e3e2b9aae645696f77571971564e4f4